scipy/linalg/flapack.pyf.src

```diff
@@ -2788,6 +2788,100 @@ subroutine <prefix2>sbevx(ab,ldab,compute_v,range,lower,n,kd,q,ldq,vl,vu,il,iu,a
+  integer intent(out) :: m
+  integer intent(hide) :: nsplit
+  <ftype2> dimension(n),depend(n),intent(out) :: w
+  integer dimension(n),depend(n),intent(out) :: iblock
+  integer dimension(n),depend(n),intent(out) :: isplit
+  integer intent(out) :: info
+end subroutine <prefix2>stebz
+
+
+subroutine <prefix2>sterf(d,e,n,info)
+  ! computes all eigenvalues of a real, symmetric tridiagonal matrix.
+
+  callstatement (*f2py_func)(&n,d,e,&info)
+  callprotoargument int*,<ctype2>*,<ctype2>*,int*
+
+  <ftype2> dimension(n),intent(in,out,copy,out=vals) :: d
+  <ftype2> dimension(n-1),intent(in,copy) :: e
+  integer depend(d),intent(hide) :: n = shape(d,0)
+  integer intent(out) :: info
+end subroutine <prefix2>sterf
+
+
+subroutine <prefix2>stein(d,e,w,iblock,isplit,m,n,z,ldz,work,iwork,ifail,info)
+  ! computes eigenvectors corresponding to eigenvalues of a real, symmetric
+  ! tridiagonal matrix.
+
+  callstatement (*f2py_func)(&n,d,e,&m,w,iblock,isplit,z,&ldz,work,iwork,ifail,&info)
+  callprotoargument int*,<ctype2>*,<ctype2>*,int*,<ctype2>*,int*,int*,<ctype2>*,int*,<ctype2>*,int*,int*,int*
+
+  <ftype2> dimension(n),intent(in) :: d
+  <ftype2> dimension(n-1),intent(in) :: e
+  <ftype2> dimension(m),intent(in) :: w
+  integer depend(w),intent(hide) :: m = shape(w,0)
+  integer depend(d),intent(hide),check(n>0) :: n = shape(d,0)
+  integer dimension(n),intent(in) :: iblock
+  integer dimension(n),intent(in) :: isplit
+  <ftype2> dimension(ldz,m),intent(out) :: z
+  integer depend(n),intent(hide) :: ldz = n
+  <ftype2> dimension(5*n),intent(hide) :: work
+  integer dimension(n),intent(hide) :: iwork
+  integer dimension(n),intent(hide) :: ifail  ! really dimension(m), but not known ahead of time
+  integer intent(out) :: info
+end subroutine <prefix2>stein
+
+
+subroutine <prefix2>stemr(d,e,range,vl,vu,il,iu,compute_v,n,m,w,z,ldz,nzc,isuppz,tryrac,work,lwork,iwork,liwork,info)
+  ! computes all eigenvalues of a real, symmetric tridiagonal matrix.
+
+  callstatement (*f2py_func)((compute_v?"V":"N"),(range>0?(range==1?"V":"I"):"A"),&n,d,e,&vl,&vu,&il,&iu,&m,w,z,&ldz,&nzc,isuppz,&tryrac,work,&lwork,iwork,&liwork,&info)
+  callprotoargument char*,char*,int*,<ctype2>*,<ctype2>*,<ctype2>*,<ctype2>*,int*,int*,int*,<ctype2>*,<ctype2>*,int*,int*,int*,int*,<ctype2>*,int*,int*,int*,int*
+
+  <ftype2> dimension(n),intent(in,copy) :: d
+  <ftype2> dimension(n),intent(in) :: e
+  integer intent(in) :: range
+  <ftype2> intent(in) :: vl
+  <ftype2> intent(in) :: vu
+  integer intent(in) :: il
+  integer intent(in) :: iu
+  integer optional,intent(in) :: compute_v = 1
+  integer intent(hide),depend(d),check(n>0) :: n = shape(d,0)
+  integer intent(out) :: m
+  <ftype2> dimension(n),depend(n),intent(out) :: w
+  <ftype2> dimension(n,n),depend(n),intent(out) :: z
+  integer depend(n),intent(hide) :: ldz = (compute_v?n:1)  ! could be made more efficient for index queries
+  integer depend(n),intent(hide) :: nzc = n
```

Comment:
The documentation mentions that `nzc==-1` is a possibility as in `_lwork` routine. Should we add it here?

Reply:
I'd assume YAGNI and let someone who needs those return values add it

Reply:
Fine with me. Actually I don't know any SciPy guidelines about these issues. So I'll leave it to the core team.